presto-spi/src/main/java/com/facebook/presto/spi/procedure/Procedure.java

```diff
@@ -91,20 +90,20 @@ public String toString()
     public static class Argument
     {
         private final String name;
-        private final Type type;
+        private final String type;
 
-        public Argument(String name, Type type)
+        public Argument(String name, String type)
```

Comment:
It seemed simpler this way, though you do get parse errors up front with that. I can change the field and leave the string constructor for convenience.